sdcm/nemesis.py

```diff
@@ -2734,11 +2734,31 @@ def _disrupt_grow_shrink_cluster(self, rack=0):
                             end=end_time, duration=time_elapsed, **log_info).publish()
 
         self._set_current_disruption("ShrinkCluster")
-        self.log.info("Start shrink cluster on %s nodes", add_nodes_number)
+        start_time = time.time()
+        # Check that number of nodes is enough for decommission:
+        cur_num_nodes_in_dc = len([n for n in self.cluster.nodes if n.dc_idx == self.target_node.dc_idx])
+        initial_db_size = self.tester.params.get("n_db_nodes")
+        if isinstance(initial_db_size, int):
+            decommission_nodes_number = min(cur_num_nodes_in_dc - initial_db_size, add_nodes_number)
```

Comment:
another thing that might be "useful", but would require to refactor this checks is to calculate as the follow:
```
if cur_num_nodes_in_dc - add_nodes_number > initial_db_size:
then run the decommission
```